src/battle_main.c

```diff
@@ -3160,6 +3216,12 @@ void FaintClearSetData(void)
     UndoFormChange(gBattlerPartyIndexes[gActiveBattler], GET_BATTLER_SIDE(gActiveBattler), FALSE);
     if (GetBattlerSide(gActiveBattler) == B_SIDE_PLAYER)
         UndoMegaEvolution(gBattlerPartyIndexes[gActiveBattler]);
+    
+    // If the fainted Pokemon was being held by Sky Drop, clear their Sky Drop data.
+    if (gActiveBattler == gBattleStruct->skyDropTargets[1])
```

Comment:
should be able to just clear `gBattleStruct->skyDropTargets[gActiveBattler]` and `gBattleStruct->skyDropTargets[gBattleStruct->skyDropTargets[gActiveBattler]]`